### ISSUE TITLE
App does not load in companion if unfilled socket in event handler 

### ISSUE THREAD

**Issue by marksherman** [mit-cml/appinventor-sources#395]
If I have an open socket, like this:
![screen shot 2015-05-28 at 11 03 37 am](https://cloud.githubusercontent.com/assets/2035149/7863258/51d9b51e-0529-11e5-8090-87db91ccb973.png)

and try to connect with the companion, it does not work. Specifically:
- After scanning a code, the code window on the computer immediately disappears (good)
- After scanning a code, the companion app does nothing. It does not issue an error, it stays at the "enter code" initial screen.
- At that time, the following error appears in the javascript console:

```
processRetVals: {"status":"BAD","type":"return","value":"java.lang.RuntimeException: invalid syntax in eval form:\n<string>:1:3275: call to 'set-and-coerce-property!' has too few arguments (3; must be 4)\n","blockid":"-1"}
```

If I try to connect over USB, the behavior is slightly different. It does connect, and it brings up a blank white app screen. The same js error appears, and nothing appears on the phone.

When I was first testing, it was a fresh/new app that had never been connected before, so the error sticker was not yet on the block. This did not seem to matter, the behavior was the same with or without the error identified in the blocks editor.

Once the app is started, the code updater seems to handle ignoring the incomplete block correctly. But on startup it does not.

**Comment by marksherman** [comment]
Seems related to #394 , and discovered at about the same time.

**Comment by halatmit** [comment]
I just tried  another example, and I looks like App inv now does not have the ability to detect empty slots at all :(

**Comment by ewpatton** [comment]
Fixed by #1895
